test/test_dynamic_shapes.py

```diff
@@ -0,0 +1,292 @@
+import torch
+from torch.utils._pytree import tree_map
+from contextlib import contextmanager
+aten = torch.ops.aten
+import unittest
+
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.testing._internal.jit_utils import JitTestCase
+
+try:
+    import sympy
+    HAS_SYMPY = True
+except ImportError:
+    HAS_SYMPY = False
+skipIfNoSympy = unittest.skipIf(not HAS_SYMPY, "no sympy")
+
+from torch._C import _disabled_torch_function_impl
+
+@contextmanager
+def no_dispatch():
+    guard = torch._C._DisableTorchDispatch()  # type: ignore[attr-defined]
+    try:
+        yield
+    finally:
+        del guard
+
+meta_funcs = {}
+
+def register_meta(op):
+    def decorator(f):
+        def add_func(op):
+            meta_funcs[op] = f
+        tree_map(add_func, op)
+        return f
+    return decorator
+
+@register_meta([aten.add.Tensor, aten.sub.Tensor])
+def binary_meta(a, b):
+    return a.new_empty(a.shape)
```

Comment:
We will need a long-term plan for these. In general if prim torch has python meta-implementations we should preferentially use those not the C++ ones.

Reply:
👍 these are essentially text fixtures to test/approximate AOTAutograd flow with symints.